### PR TITLE
ocamlPackages.reason-native: 2022-08-31-a0ddab6 -> <version>-unstable-2024-05-07

### PR DESCRIPTION
**Pull request by philiptaron**
## Description of changes

Upstream doesn't use releases, branches, or tags, so this is the most recent commit: https://github.com/reasonml/reason-native/commit/20b1997b6451d9715dfdbeec86a9d274c7430ed8

* ocamlPackages.reason-native.cli: init at 0.0.1-alpha-unstable-2024-05-07
* ocamlPackages.reason-native.console: 2022-08-31-a0ddab6 -> 0.1.0-unstable-2024-05-07
* ocamlPackages.reason-native.dir: 2022-08-31-a0ddab6 -> 0.0.1-unstable-2024-05-07
* ocamlPackages.reason-native.file-context-printer: 2022-08-31-a0ddab6 -> 0.0.3-unstable-2024-05-07
* ocamlPackages.reason-native.fp: 2022-08-31-a0ddab6 -> 0.0.1-unstable-2024-05-07
* ocamlPackages.reason-native.frame: init at 0.0.1-unstable-2024-05-07
* ocamlPackages.reason-native.fs: init at 0.0.2-unstable-2024-05-07
* ocamlPackages.reason-native.pastel: 2022-08-31-a0ddab6 -> 0.3.0-unstable-2024-05-07
* ocamlPackages.reason-native.pastel-console: 2022-08-31-a0ddab6 -> 0.0.0-unstable-2024-05-07
* ocamlPackages.reason-native.qcheck-rely: 2022-08-31-a0ddab6 -> 1.0.2-unstable-2024-05-07
* ocamlPackages.reason-native.refmterr: 2022-08-31-a0ddab6 -> 3.3.0-unstable-2024-05-07
* ocamlPackages.reason-native.rely: 2022-08-31-a0ddab6 -> 4.0.0-unstable-2024-05-07
* ocamlPackages.reason-native.rely-junit-reporter: 2022-08-31-a0ddab6 -> 1.0.0-unstable-2024-05-07
* ocamlPackages.reason-native.unicode: init at 0.0.0-unstable-2024-05-07
* ocamlPackages.reason-native.unicode-config: init at 0.0.0-unstable-2024-05-07
* ocamlPackages.reason-native.utf8: init at 0.1.0-unstable-2024-05-07

The version numbers are taken from upstream's `<package>.json` file.

The constituent packages were in an almost-but-not-quite `callPackage` style. I've updated them so that they're all actual packages. There's a mild amount of duplication induced by doing so, but no more so than every package in `pkgs/by-name`.

I also fixed and updated `ocamlPackages.reason-native.console.tests` so that it builds and no longer includes its `default.nix` file.

- #301014

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/reason-native/cli.nix [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
+{ lib, buildDunePackage, re, reason, pastel, src }:
+
+buildDunePackage {
+  inherit src;
 
-{
   pname = "cli";
+  version = "0.0.1-alpha-unstable-2024-05-07";
+
+  duneVersion = "3";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/reason-native/cli.nix:9]: I think you can remove all these `duneVersion` attributes.

**Comment by superherointj** [comment]
> Upstream doesn't use releases, branches, or tags

When I initially packaged this I wasn't aware of:

> passthru.updateScript = unstableGitUpdater { };

Can be executed as:

> nix-shell maintainers/scripts/update.nix --argstr package ocamlPackages.reason-native.paste

Things need to be tweaked a bit. But it could prove helpful in case you want to keep it fresh, not that upstream is changing much.

File: pkgs/development/ocaml-modules/reason-native/cli.nix (hunk 1 above)
Comment (philiptaron) [reply pkgs/development/ocaml-modules/reason-native/cli.nix:9]: Right you are. From `pkgs/build-support/ocaml/dune.nix`:

```
let Dune =
  let dune-version = args.duneVersion or "3"; in
  { "1" = dune_1; "2" = dune_2; "3" = dune_3; }."${dune-version}"
; in
```

**Comment by philiptaron** [comment]
> Things need to be tweaked a bit. But it could prove helpful in case you want to keep it fresh, not that upstream is changing much.

Thanks for the link and the information. I thought about how updates would happen, given that this update was a little manual... but that slow rate of changes made me think it wasn't going to be an issue. If I'm in the area again in another couple years, let's see if I think differently.

**Pull request merged**